cmd/version.go

```diff
@@ -47,10 +49,54 @@ type version struct {
 
 // versionString returns the CLI version
 func versionString() string {
-	if kubeBuilderVersion == unknown {
-		if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "" {
+	info, ok := debug.ReadBuildInfo()
+
+	if ok && info.Main.Version != "" {
+		if kubeBuilderVersion == unknown {
 			kubeBuilderVersion = info.Main.Version
 		}
+
+		if goos == unknown {
+			goos = runtime.GOOS
+		}
+
+		if goarch == unknown {
+			goarch = runtime.GOARCH
+		}
+
+		if gitCommit == unknown && info.Main.Version != "" {
+			mainVersionSplit := strings.Split(info.Main.Version, "-")
+
+			// For released semvers like "v4.5.0"
+			// Result: info.Main.Version == "semver"
+			if len(mainVersionSplit) == 1 {
+				gitCommit = info.Main.Version
```

Comment:
It should be validated, but probably much easy to get commit hash (vcs.revision) and time (vcs.time) from the `info.Settings` like
```go
for _, setting := range info.Settings {
		if setting.Key == "vcs.revision" {
			gitCommit = setting.Value
		}
	}
```